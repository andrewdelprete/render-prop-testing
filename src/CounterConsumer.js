import React from "react";
import Counter from "./Counter";

export default function CounterConsumer({ initial }) {
  return (
    <Counter initial={initial}>
      {({ increment, decrement, counter }) => (
        <div className="content" style={{ textAlign: "center" }}>
          <h1>{counter}</h1>
          <button className="button is-success" onClick={increment}>
            Increment
          </button>
          <button className="button is-danger" onClick={decrement}>
            Decrement
          </button>
        </div>
      )}
    </Counter>
  );
}
