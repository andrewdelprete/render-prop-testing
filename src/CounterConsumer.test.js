import React from "react";
import ReactDOM from "react-dom";
import toJSON from "enzyme-to-json";
import { mount } from "enzyme";
import "./enzymeSetup";
import CounterConsumer from "./CounterConsumer";

it("accepts an initial value", () => {
  const wrapper = mount(<CounterConsumer initial={1999} />);
  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("increments counter", () => {
  const wrapper = mount(<CounterConsumer />);
  wrapper
    .find("button")
    .at(0)
    .simulate("click");

  expect(toJSON(wrapper)).toMatchSnapshot();
});

it("decrements counter", () => {
  const wrapper = mount(<CounterConsumer />);
  wrapper
    .find("button")
    .at(1)
    .simulate("click");

  expect(toJSON(wrapper)).toMatchSnapshot();
});
