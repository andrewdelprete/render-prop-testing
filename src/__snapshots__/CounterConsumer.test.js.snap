// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts an initial value 1`] = `
<CounterConsumer
  initial={1999}
>
  <Counter
    initial={1999}
  >
    <div
      className="content"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h1>
        1999
      </h1>
      <button
        className="button is-success"
        onClick={[Function]}
      >
        Increment
      </button>
      <button
        className="button is-danger"
        onClick={[Function]}
      >
        Decrement
      </button>
    </div>
  </Counter>
</CounterConsumer>
`;

exports[`decrements counter 1`] = `
<CounterConsumer>
  <Counter>
    <div
      className="content"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h1>
        -1
      </h1>
      <button
        className="button is-success"
        onClick={[Function]}
      >
        Increment
      </button>
      <button
        className="button is-danger"
        onClick={[Function]}
      >
        Decrement
      </button>
    </div>
  </Counter>
</CounterConsumer>
`;

exports[`increments counter 1`] = `
<CounterConsumer>
  <Counter>
    <div
      className="content"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h1>
        1
      </h1>
      <button
        className="button is-success"
        onClick={[Function]}
      >
        Increment
      </button>
      <button
        className="button is-danger"
        onClick={[Function]}
      >
        Decrement
      </button>
    </div>
  </Counter>
</CounterConsumer>
`;
